src/main/java/org/generationcp/commons/ruleengine/ExpressionUtils.java

```diff
@@ -0,0 +1,52 @@
+package org.generationcp.commons.ruleengine;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class ExpressionUtils {
+
+	public static final Integer DEFAULT_LENGTH = 3;
+
+	private ExpressionUtils() {
+		// utility class
+	}
+
+	public static void replaceExpressionWithValue(final String expressionKey, final StringBuilder container, final String value) {
+		final int startIndex = container.toString().toUpperCase().indexOf(expressionKey);
+		final int endIndex = startIndex + expressionKey.length();
+
+		final String replaceValue = value == null ? "" : value;
+		container.replace(startIndex, endIndex, replaceValue);
+	}
+
+	public static void replaceRegularExpressionKeyWithValue (final String expressionKey, final StringBuilder container, final String value) {
+		final Pattern pattern = Pattern.compile(expressionKey);
```

Comment:
For performance improvement, I think it's better to pass the Pattern itself, and compile the regex once.

Reply:
Addressed this @abatac